File: selector/src/main/java/com/luck/picture/lib/dialog/PictureLoadingDialog.java
package com.luck.picture.lib.dialog;


import android.app.Dialog;
import android.content.Context;
import android.os.Bundle;
import android.view.Gravity;
import android.view.ViewGroup;
import android.view.WindowManager;

import com.luck.picture.lib.R;

public class PictureLoadingDialog extends Dialog {

    public PictureLoadingDialog(Context context) {
        super(context, R.style.Picture_Theme_AlertDialog);
        setCancelable(true);
        setCanceledOnTouchOutside(false);
    }

    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.ps_alert_dialog);
        setDialogSize();
    }

    private void setDialogSize() {
        WindowManager.LayoutParams params = getWindow().getAttributes();
        params.width = ViewGroup.LayoutParams.WRAP_CONTENT;
        params.height = ViewGroup.LayoutParams.WRAP_CONTENT;
        params.gravity = Gravity.CENTER;
        getWindow().setWindowAnimations(R.style.PictureThemeDialogWindowStyle);
        getWindow().setAttributes(params);
    }
}

File: selector/src/main/java/com/luck/picture/lib/dialog/PictureLoadingDialog.kt
package com.luck.picture.lib.dialog

import android.app.Dialog
import android.content.Context
import android.view.Gravity
import android.view.ViewGroup
import com.luck.picture.lib.R

/**
 * @author：luck
 * @date：2021/11/17 10:24 上午
 * @describe：loading dialog
 */
class PictureLoadingDialog(context: Context) : Dialog(context, R.style.Picture_Theme_AlertDialog) {

    private fun setDialogSize() {
        window?.attributes?.apply {
            this.width = ViewGroup.LayoutParams.WRAP_CONTENT
            this.height = ViewGroup.LayoutParams.WRAP_CONTENT
            this.gravity = Gravity.CENTER
            window?.setWindowAnimations(R.style.PictureThemeDialogWindowStyle)
            window?.attributes = this
        }
    }

    init {
        setContentView(R.layout.ps_alert_dialog)
        setCancelable(true)
        setCanceledOnTouchOutside(false)
        setDialogSize()
    }
}